src/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -111,6 +111,30 @@ protected byte[] GetBuffer(int count, out int offset)
             return _buffer;
         }
 
+        protected async Task<Tuple<byte[], int>> GetBufferAsync(int count)
```

Comment:
I don't know whether it matters for the expected scenarios, but every call to this method is going to require allocating a tuple object; you could avoid that by using a struct type instead.